wazimap/data/tables.py

```diff
@@ -547,13 +559,15 @@ def permute(level, field_keys, rows):
 
         return data
 
-    def _build_model_from_fields(self, fields, table_name, geo_level=None, value_type=Integer):
+    def _build_model_from_fields(self, fields, table_name, geo_level=None, value_type=Integer, existing_table=None):
```

Comment:
What if we just pass in ``existing_table`` as ``table_name``? That way it'll return the cached model (hopefully that works ok).